### PR TITLE
Add hash field for Transaction and fix evm tx id.

### PR DESCRIPTION
**Pull request by shaitao**
Hash of raw tx can be delivered now.

* **make sure that you have executed all the following process and no errors occur**
  - [x] make fmt
  - [ ] make lint
  - [x] make test

* **The major changes of this PR**

Add a `hash` field for `Transaction`

Hash of raw tx can be delivered when "contract-withdraw".

* **Extra documentations**

Upgrade height is needed because old `Transaction` don't have field `hash`.

### CODE REVIEW

File: src/components/contracts/baseapp/src/lib.rs [hunk 1]
```diff
@@ -301,7 +301,9 @@ impl BaseApp {
         self.modules.process_findora_tx(&self.deliver_state, tx)
     }
 
-    pub fn consume_mint(&mut self) -> Option<Vec<NonConfidentialOutput>> {
+    pub fn consume_mint(
+        &mut self,
+    ) -> Option<(Vec<NonConfidentialOutput>, Option<String>)> {
```
Comment (ws4charlie) [inline src/components/contracts/baseapp/src/lib.rs:306]: (Vec<NonConfidentialOutput>, Option<String>) means 1 tx hash produces multiple UTXOs. How are we handling multiple internal transactions (in same block) in here(EVM --> UTXO).
Shouldn't be (Vec<Vec<NonConfidentialOutput>>, Vec<Option<String>>) or 
                       (Vec<Vec<NonConfidentialOutput>>, Vec<String>) ?? (bcs hash must exist in this case).

Comment (ws4charlie) [reply src/components/contracts/baseapp/src/lib.rs:306]: @tiannian

Comment (ws4charlie) [reply src/components/contracts/baseapp/src/lib.rs:306]: @shaitao can you take a look at this

Comment (shaitao) [reply src/components/contracts/baseapp/src/lib.rs:306]: According to the definition of `Transaction`,  it has a `Vec<Operation>`, it's ok to produce multiple UTXO in a transaction, but one transaction have only one hash. Multiple transactions is deliverd by tendermint sequentially.

**Pull request closed without merging**